From 6da19e5b73005345fc693f00a3507c999be9644f Mon Sep 17 00:00:00 2001
From: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Date: Wed, 10 Jul 2013 08:57:03 +0200
Subject: [PATCH 186/429] wifi: Enable Wifi on SaltBay LnP

Enable iwlwifi platform data

Category: device enablement
Domain: CWS-Wifi
Origin: internal
Upstream-Candidate: no, not-aosp

Signed-off-by: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Signed-off-by: remi laudebat <remi.laudebat@intel.com>
Signed-off-by: Mick CORNUT <mick.cornut@intel.com>
Signed-off-by: Jeremie GARCIA <jeremie.garcia@intel.com>
Signed-off-by: Frode Isaksen <frodex.isaksen@intel.com>
---
 drivers/mmc/host/sdhci.c     | 2 ++
 include/linux/mmc/sdhci.h    | 2 ++
 include/linux/mmc/sdio_ids.h | 2 ++
 3 files changed, 6 insertions(+)

diff --git a/drivers/mmc/host/sdhci.c b/drivers/mmc/host/sdhci.c
index 0d23c3567e2e..d1e8eda47dcd 100644
--- a/drivers/mmc/host/sdhci.c
+++ b/drivers/mmc/host/sdhci.c
@@ -3109,6 +3109,8 @@ int sdhci_add_host(struct sdhci_host *host)
 
 	if (host->quirks2 & SDHCI_QUIRK2_ADVERTISE_2V0_FORCE_1V8)
 		ocr_avail |= MMC_VDD_20_21;
+	if (host->quirks2 & SDHCI_QUIRK2_ADVERTISE_3V0_FORCE_1V8)
+		ocr_avail |= MMC_VDD_32_33;
 
 	mmc->ocr_avail = ocr_avail;
 	mmc->ocr_avail_sdio = ocr_avail;
diff --git a/include/linux/mmc/sdhci.h b/include/linux/mmc/sdhci.h
index 7f948641707a..daf7d693e0f8 100644
--- a/include/linux/mmc/sdhci.h
+++ b/include/linux/mmc/sdhci.h
@@ -118,6 +118,8 @@ struct sdhci_host {
 #define SDHCI_QUIRK2_WAIT_FOR_IDLE			(1<<25)
 /* BAD sd cd in HOST IC. This will cause system hang when removing SD */
 #define SDHCI_QUIRK2_BAD_SD_CD				(1<<26)
+#define SDHCI_QUIRK2_POWER_PIN_GPIO_MODE		(1<<27)
+#define SDHCI_QUIRK2_ADVERTISE_3V0_FORCE_1V8   (1<<28)
 
 	int irq;		/* Device IRQ */
 	void __iomem *ioaddr;	/* Mapped address */
diff --git a/include/linux/mmc/sdio_ids.h b/include/linux/mmc/sdio_ids.h
index 9f03feedc8e7..b3933db80fa9 100644
--- a/include/linux/mmc/sdio_ids.h
+++ b/include/linux/mmc/sdio_ids.h
@@ -23,6 +23,8 @@
 /*
  * Vendors and devices.  Sort key: vendor first, device next.
  */
+#define IWL_SDIO_DEVICE_ID_WKP1	0x3160
+#define IWL_SDIO_DEVICE_ID_WKP2	0x7260
 #define SDIO_VENDOR_ID_INTEL			0x0089
 #define SDIO_DEVICE_ID_INTEL_IWMC3200WIMAX	0x1402
 #define SDIO_DEVICE_ID_INTEL_IWMC3200WIFI	0x1403
-- 
2.37.3

